lib/Sema/TypeCheckPattern.cpp

```diff
@@ -475,15 +400,18 @@ class ResolvePattern : public ASTVisitor<ResolvePattern,
   // Member syntax 'T.Element' forms a pattern if 'T' is an enum and the
   // member name is a member of the enum.
   Pattern *visitUnresolvedDotExpr(UnresolvedDotExpr *ude) {
-    SmallVector<IdentTypeRepr *, 2> components;
-    if (!ExprToDeclRefTypeRepr(components, Context).visit(ude->getBase()))
+    // In synthesized code this will have been a pattern from the get-go.
```

Comment:
What do you mean by “from the get-go”?

Reply:
I was trying to express that we are expected to be smart enough not to synthesize code that would need this kind of simplification. That is, you wouldn‘t build an expression for sth that you know is an enum element pattern.

Reply:
Oh, I just wasn’t familiar wit the phrase “from the get-go”. Makes sense